async-nats/src/lib.rs

```diff
@@ -981,6 +983,30 @@ pub async fn connect_with_options<A: ToServerAddrs>(
             connect_info.user = Some(user);
             connect_info.pass = Some(pass);
         }
+        Authorization::NKey(seed) => {
+            match nkeys::KeyPair::from_seed(str::from_utf8(seed.unsecure()).unwrap())
+                .map_err(AuthError::new)
+            {
+                Ok(key_pair) => {
+                    let nonce = server_info.nonce.clone();
+                    match key_pair.sign(nonce.as_bytes()).map_err(AuthError::new) {
```

Comment:
This approach works for the initial connection, but it will fail for any reconnections, as when reconnecting, you have to sign a new nonce challenge.

A simple implementation is to store the NKEY pair in memory and sign the nonce whenever reconnection happens.

A more secure one is to lazy load the `seed` every time client reconnects.

I would go for the simple approach as the first pass, especially as we want to add lazy loading to more auth methods.

Reply:
yes, exactly. Let me know if it worked quickly and well. I can fix it before working with NKEY PR in case of any problems, as the problem is not caused by this PR obviously.

Reply:
Ah ok.
I don't have a solution coded yet, so go ahead!